### PR TITLE
#7 Task. Punko Pavel

### PR DESCRIPTION
**Pull request by PashaPunko**
https://travis-ci.org/github/PashaPunko/js-assignments/builds/706572211

### CODE REVIEW

File: task/07-yield-tasks.js [hunk 1]
```diff
@@ -108,7 +134,22 @@ function* depthTraversalTree(root) {
  *
  */
 function* breadthTraversalTree(root) {
```
Comment (AlexMokin) [inline task/07-yield-tasks.js:136]: thre is a way to use javascript array as a queue. See the shift method. Try to refactor this with 1 arrray

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift

**Review by AlexMokin — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/07-yield-tasks.js (hunk 1 above)
Comment (PashaPunko) [reply task/07-yield-tasks.js:136]: shift method works too slow, so the solution didn't pass the time limit test

**Comment by AlexMokin** [comment]
accepted

**Pull request closed without merging**